src/tools/replace-version-placeholder/src/main.rs

```diff
@@ -0,0 +1,49 @@
+use std::path::PathBuf;
+
+macro_rules! t {
+    ($e:expr, $p:expr) => {
+        match $e {
+            Ok(e) => e,
+            Err(e) => panic!("{} failed on {} with {}", stringify!($e), ($p).display(), e),
+        }
+    };
+
+    ($e:expr) => {
+        match $e {
+            Ok(e) => e,
+            Err(e) => panic!("{} failed with {}", stringify!($e), e),
+        }
+    };
+}
+
+pub const VERSION_PLACEHOLDER: &str = "CURRENT_RUSTC_VERSION";
+
+#[path = "../../tidy/src/walk.rs"]
+#[allow(dead_code)]
+pub mod walk;
```

Comment:
This seems pretty surprising to me. Maybe we should move this tool into tidy (with a --bless flag of some kind) or otherwise share the code in a way that's not sharing by path attribute?